cypress/integration/sales/void_a_sales_invoice_then_invoice_the_billing_candidates_again.js

```diff
@@ -0,0 +1,372 @@
+/*
+ * #%L
+ * metasfresh-e2e
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+import { DocumentStatusKey, DocumentActionKey } from '../../support/utils/constants';
+import { BillingCandidates } from '../../page_objects/billing_candidates';
+import { salesOrders } from '../../page_objects/sales_orders';
+import { ShipmentDispositions } from '../../page_objects/shipment_dispositions';
+import { SalesOrder, SalesOrderLine } from '../../support/utils/sales_order';
+import { getLanguageSpecific } from '../../support/utils/utils';
+import { salesInvoices } from '../../page_objects/sales_invoices';
+
+describe('Void Sales Invoice and invoice the billing candidates again', function() {
+  const businessPartnerName = 'Test Lieferant 1';
+  const productName = 'Convenience Salat 250g';
+  const originalQuantity = 200;
+  const shipmentNotificationInboxText = `Lieferung [number] für Partner ${businessPartnerName} wurde erstellt.`;
+  const shipmentNotificationModalText = 'Created: 1 WorkPackage Queue;';
+  const generateInvoicesNotificationModalText =
+    'Fakturlauf mit 1 Rechnungen eingeplant. Es sind bereits 0 zu erstellende Rechnungen in der Warteschlange, die vorher verarbeitet werden.';
+
+  const salesInvoiceDocumentType = 'Sales Invoice';
+
+  // for test
+  let salesOrderDocumentNumber;
+  let salesOrderRecordId;
+  let totalAmountToPay = 0;
+
+  before(function() {
+    // This wait is stupid.
+    // It also appears to be a good workaround for the problems in
+    // cypress/support/utils/utils.js:1
+    cy.wait(5000);
```

Comment:
we'll have to fix that at some point

Reply:
I tried to figure out a way to get the "time since the test started" and use that to compare inside `getLanguageSpecific` and do the sleep there (if there's less that 5 seconds since the start), but i didn't find any. If you know of such a thing just ping me and i can work on it from there.

Reply:
but what is it actually doing, that we have to run it before all tests ?

Reply:
It's not before all, only those tests which call in the first step the function `getLanguageSpecific`, which i am doing just a few lines below @ `.setDocumentAction(getLanguageSpecific(miscDictionary, DocumentActionKey.Complete))`